### ISSUE TITLE
JSX opening element closing tag gets moved to a new line

### ISSUE THREAD

**Issue by gcazaciuc** [jlongster/prettier#654]
If you have a line of JSX that is over printWidth, something like:

```js
const El = (<span width="1" height="2">x</span>)
```

For the sake of reproducing the issue assume the printWidth is set to 20 but this reproduces anytime the line needs to break for JSX. Then the code is formatted as:

```js
const El = (
    <span
        width="1"
        height="2"
    >
        x
    </span>
);
```

Notice the span `>` being placed on a new line by itself. 
Again this wastes lots of vertical space whenever you have many tags in a component. Instead a better option would be either to make it configurable or to place it on the same line as the last prop, eg:

```js
const El = (
    <span
        width="1"
        height="2" >
        x
    </span>
);
```

**Comment by jlongster** [comment]
This is intentional, but there have been [discussions](https://github.com/jlongster/prettier/issues/467#issuecomment-275230393) before about it. There's a PR where we decided to do it but I can't find it.

The second code block in my opinion is a log harder to read. The attributes line up perfectly with `x` and it's very difficult to disambiguate when the opening element ends. This is why we chose to moving it to a new line as it's more consistent.

I think this style is the future of JSX as a lot more people are moving towards it. However, there is a lot of code out there right now with the your style (including Facebook) so we'll probably end up making this an option. We haven't added many options but this one has come up enough that we probably will.
